staging/src/k8s.io/apiserver/pkg/server/routes/flags.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routes
+
+import (
+	"html/template"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	"k8s.io/apiserver/pkg/server/mux"
+)
+
+var (
+	lock          = &sync.RWMutex{}
+	registedFlags = map[string]debugFlag{}
+)
+
+// Flags adds handlers for flags under /debug/flags.
+type Flags struct {
+}
+
+// Install registers the APIServer's flags handler.
+func (f Flags) Install(c *mux.PathRecorderMux, flag string, handler func(http.ResponseWriter, *http.Request)) {
+	c.UnlistedHandle("/debug/flags", http.HandlerFunc(f.Index))
+	c.UnlistedHandlePrefix("/debug/flags/", http.HandlerFunc(f.Index))
+
+	url := path.Join("/debug/flags", flag)
+	c.UnlistedHandleFunc(url, handler)
+
+	f.addFlag(flag)
+}
+
+// Index responds with the `/debug/flags` request.
+// For example, "/debug/flags/v" serves the "--v" flag.
+// Index responds to a request for "/debug/flags/" with an HTML page
+// listing the available flags.
+func (f Flags) Index(w http.ResponseWriter, r *http.Request) {
+	lock.RLock()
+	defer lock.RUnlock()
+	if err := indexTmpl.Execute(w, registedFlags); err != nil {
+		glog.Error(err)
+	}
+}
+
+var indexTmpl = template.Must(template.New("index").Parse(`<html>
+<head>
+<title>/debug/flags/</title>
+</head>
+<body>
+/debug/flags/<br>
+<br>
+flags:<br>
+<table>
+{{range .}}
+<tr>{{.Flag}}<br>
+{{end}}
+</table>
+<br>
+full flags configurable<br>
+</body>
+</html>
+`))
+
+type debugFlag struct {
+	Flag string
+}
+
+func (f Flags) addFlag(flag string) {
+	lock.Lock()
+	defer lock.Unlock()
+	registedFlags[flag] = debugFlag{flag}
+}
+
+func (f Flags) getFlags(flag string) []debugFlag {
+	flags := []debugFlag{}
+	lock.RLock()
+	defer lock.RUnlock()
+	for _, val := range registedFlags {
+		flags = append(flags, val)
+	}
+	return flags
+}
+
+// VerbosityHandler is the HTTP handler for setting glog `-v` flag.
+func VerbosityHandler(w http.ResponseWriter, req *http.Request) {
+	var gloglevel glog.Level
+	switch {
+	case req.Method == "PUT":
+		defer req.Body.Close()
+		body, _ := ioutil.ReadAll(req.Body)
+		if err := gloglevel.Set(string(body)); err != nil {
+			responsewriters.WriteRawJSON(http.StatusBadRequest, err, w)
+			return
+		}
+		responsewriters.WriteRawJSON(http.StatusOK, "set glog.logging.verbosity to "+string(body), w)
```

Comment:
set glog.logging.verbosity to "+string(body) is not json.

Reply:
addressed